odoo/models.py

```diff
@@ -5703,6 +5703,16 @@ def __iter__(self):
             for id_ in self._ids:
                 yield self.__class__(self.env, (id_,), self._prefetch_ids)
 
+    def __reversed__(self):
+        """ Return an reversed iterator over ``self``. """
+        if len(self._ids) > PREFETCH_MAX and self._prefetch_ids is self._ids:
+            for ids in self.env.cr.split_for_in_conditions(reversed(self._ids)):
+                for id_ in ids:
+                    yield self.__class__(self.env, (id_,), ids)
+        else:
+            for id_ in reversed(self._ids):
+                yield self.__class__(self.env, (id_,), reversed(tuple(self._prefetch_ids)))
```

Comment:
This iterates over `self._prefetch_ids`, which you really want to avoid :frowning_face:

Reply:
Maybe we should reimplement the iterables used for `_prefetch_ids` on relational fields to let them implement `__reversed__`.

Reply:
I didn't to modify too much. But I am ok with it, I will change the _prefetch_ids of relational fields to be reversible.

Reply:
What I meant is to turn the helper functions `prefetch_many2one_ids` into actual classes that implement `__iter__` and `__reversed__`...

Reply:
Yep. Sorry, my comment was unclear (forget a word): "I didn't WANT to modify too much code"